app/controllers/api/generic_object_definitions_controller.rb

```diff
@@ -98,13 +98,13 @@ def remove_methods_resource(type, id, data)
     end
 
     def self.allowed_association_types
-      GenericObjectDefinition::ALLOWED_ASSOCIATION_TYPES.collect do |association_type|
-        [Dictionary.gettext(association_type, :type => :model, :notfound => :titleize, :plural => false), association_type]
-      end.sort
+      GenericObjectDefinition::ALLOWED_ASSOCIATION_TYPES.sort.map do |association_type|
+        [association_type, Dictionary.gettext(association_type, :type => :model, :notfound => :titleize, :translate => false)]
+      end.to_h
     end
 
     def self.allowed_types
-      GenericObjectDefinition::TYPE_NAMES.stringify_keys.invert.to_a.sort
+      GenericObjectDefinition::TYPE_NAMES.stringify_keys
```

Comment:
On further consideration, I think the `stringify_keys` isn't needed, since Rails will know how to convert these to valid JSON

Reply:
You're right, it's not needed (tested).